File: trunk/net.tenteCsharp.templatesProject/src_code/heaterManagement/RoomGUI.java


cclass heaterManagement.HeaterManagement; 

public cclass RoomGUI extends TypeComponent{ 

	public HeaterNotifyPort heaterNotifyPort;
		public ArrayList listThermometerGUI;
		public ArrayList listHeaterGUI;
	
	
	public RoomGUI (String id){
		super(id);
	}
	
	public HeaterNotifyPort getHeaterNotifyPort(){
		return heaterNotifyPort;
	}
	
	public void setHeaterNotifyPort(RoomGUI.HeaterNotifyPort value){
		this.heaterNotifyPort=value;
	}
	
	public ArrayList getListThermometerGUI(){
		return listThermometerGUI;
	}
	
	public void setListThermometerGUI(ArrayList value){
		this.listThermometerGUI=value;
	}
	
	public void addListThermometerGUIElement(ThermometerGUI value){
		this.listThermometerGUI.add(value);
	}
	
	public ArrayList getListHeaterGUI(){
		return listHeaterGUI;
	}
	
	public void setListHeaterGUI(ArrayList value){
		this.listHeaterGUI=value;
	}
	
	public void addListHeaterGUIElement(HeaterGUI value){
		this.listHeaterGUI.add(value);
	}
	

	
	
	public cclass HeaterNotifyPort extends TypePort{

		public ArrayList portsIGeneralHeaterNotify;

		public HeaterNotifyPort (){
			super();
			portsIGeneralHeaterNotify=new ArrayList();

		}


			
		public void connectPort(IGeneralHeaterNotify port){
			portsIGeneralHeaterNotify.add(port);
		}
	}

}

File: trunk/Elio/pseudoCodeGeneratorElio/src-gen/heaterManagement/RoomGUI.cs
using System;
using System.Collections;
using System.Text;

namespace SmartHome
{
	public partial class RoomGUI : TypeComponent
	{
		public HeaterNotifyPort heaterNotifyPort;
			public  ArrayList listThermometerGUI= new ArrayList();
			public  ArrayList listHeaterGUI= new ArrayList();
	
		public RoomGUI (String id) 
			: base(id)
		{
		}
		public HeaterNotifyPort getHeaterNotifyPort()
		{
			return heaterNotifyPort;
		}
	
		public void setHeaterNotifyPort(RoomGUI.HeaterNotifyPort value)
		{
			this.heaterNotifyPort=value;
		}
	
		public  ArrayList getListThermometerGUI()
		{
			return listThermometerGUI;
		}
	
		public void setListThermometerGUI(ArrayList value)
		{
			this.listThermometerGUI=value;
		}
		
		public void addListThermometerGUIElement(ThermometerGUI value)
		{
			this.listThermometerGUI.Add(value);
		}
	
		public  ArrayList getListHeaterGUI()
		{
			return listHeaterGUI;
		}
	
		public void setListHeaterGUI(ArrayList value)
		{
			this.listHeaterGUI=value;
		}
		
		public void addListHeaterGUIElement(HeaterGUI value)
		{
			this.listHeaterGUI.Add(value);
		}
	

	
		
		public class HeaterNotifyPort : TypePort
		{
 		public ArrayList portsIGeneralHeaterNotify = new ArrayList();

			public HeaterNotifyPort()
				: base()
			{
				portsIGeneralHeaterNotify=new ArrayList();

			}


			public ArrayList getPortsIGeneralHeaterNotify()
			{
	        	return portsIGeneralHeaterNotify;
	        }
				
			public void connectPort(IGeneralHeaterNotify port)
			{
				portsIGeneralHeaterNotify.Add(port);
			}
	
		}
	}
}